service/internal/telemetrylogs/logger.go

```diff
@@ -45,5 +46,10 @@ func NewLogger(cfg config.ServiceTelemetryLogs, options []zap.Option) (*zap.Logg
 	if err != nil {
 		return nil, err
 	}
+
+	if cfg.EnableGrpcLogs {
+		grpcZap.ReplaceGrpcLoggerV2(logger)
```

Comment:
Is it possible to do this at instance level instead of globally? we have helpers like configgrpc for this.

Reply:
Thank you @bogdandrutu , let me look into it. I was about to clarify on this same question.

Reply:
@jpkrohling @bogdandrutu it looks like the way to go about it is to chain unary and stream interceptors provided by grpc_zap. We already are doing them for tracing and authentication purposes.

This all said one question remains, do we want to use the same zap logger that we set for collector process or do we want to create a new on? please advice. If we want to handle the same zap logger then the next step would be to get access to it somehow.

Reply:
cc: @pureklkl

Reply:
@bogdandrutu @jpkrohling while it is possible to introduce logger using interceptors per each grpc call - i think the dial level logs are limited to the information from request sessions only and we do not see the logs from grpc library internals. 

for instance. if I use UnaryInterceptor for a failure case

```bash 
warn	zap/options.go:212	finished client unary call	{"system": "grpc", "span.kind": "client", "grpc.service": "protof.Product", "grpc.method": "search", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:6754: connect: connection refused\"", "grpc.code": "Unavailable", "grpc.time_ms": 0.8080000281333923}
2021/11/11 16:47:27 could not greet: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp [::1]:6754: connect: connection refused"
exit status 1
```

but setting `grpcZap.ReplaceGrpcLoggerV2(logger)` produce grpc library internals which I think is what @jpkrohling wants (correct me if I am wrong)

```bash
2021/11/11 16:49:41 overriding logger
1.636678181635326e+09	info	zap/grpclogger.go:92	[core]original dial target is: "localhost:6754"	{"system": "grpc", "grpc_log": true}
1.636678181635499e+09	info	zap/grpclogger.go:92	[core]parsed dial target is: {Scheme:localhost Authority: Endpoint:6754 URL:{Scheme:localhost Opaque:6754 User: Host: Path: RawPath: ForceQuery:false RawQuery: Fragment: RawFragment:}}	{"system": "grpc", "grpc_log": true}
1.636678181635517e+09	info	zap/grpclogger.go:92	[core]fallback to scheme "passthrough"	{"system": "grpc", "grpc_log": true}
1.636678181635544e+09	info	zap/grpclogger.go:92	[core]parsed dial target is: {Scheme:passthrough Authority: Endpoint:localhost:6754 URL:{Scheme:passthrough Opaque: User: Host: Path:/localhost:6754 RawPath: ForceQuery:false RawQuery: Fragment: RawFragment:}}	{"system": "grpc", "grpc_log": true}
1.636678181635558e+09	info	zap/grpclogger.go:92	[core]Channel authority set to "localhost:6754"	{"system": "grpc", "grpc_log": true}
1.6366781816355789e+09	info	zap/grpclogger.go:92	[core]ccResolverWrapper: sending update to cc: {[{localhost:6754  <nil> <nil> 0 <nil>}] <nil> <nil>}	{"system": "grpc", "grpc_log": true}
1.636678181635591e+09	info	zap/grpclogger.go:92	[core]ClientConn switching balancer to "pick_first"	{"system": "grpc", "grpc_log": true}
1.636678181635603e+09	info	zap/grpclogger.go:92	[core]Channel switches to new LB policy "pick_first"	{"system": "grpc", "grpc_log": true}
2021/11/11 16:49:41 creating a new client
2021/11/11 16:49:41 Sending product name: Laptop
1.636678181635682e+09	info	zap/grpclogger.go:92	[core]Subchannel Connectivity change to CONNECTING	{"system": "grpc", "grpc_log": true}
1.636678181635735e+09	info	zap/grpclogger.go:92	[core]pickfirstBalancer: UpdateSubConnState: 0xc0000207b0, {CONNECTING <nil>}	{"system": "grpc", "grpc_log": true}
1.6366781816357179e+09	info	zap/grpclogger.go:92	[core]Subchannel picks a new address "localhost:6754" to connect	{"system": "grpc", "grpc_log": true}
1.636678181635755e+09	info	zap/grpclogger.go:92	[core]Channel Connectivity change to CONNECTING	{"system": "grpc", "grpc_log": true}
1.636678181635749e+09	info	zap/grpclogger.go:92	[core]blockingPicker: the picked transport is not ready, loop back to repick	{"system": "grpc", "grpc_log": true}
1.636678181636446e+09	warn	zap/grpclogger.go:104	[core]grpc: addrConn.createTransport failed to connect to {localhost:6754 localhost:6754 <nil> <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp [::1]:6754: connect: connection refused"	{"system": "grpc", "grpc_log": true}
1.636678181636482e+09	info	zap/grpclogger.go:92	[core]Subchannel Connectivity change to TRANSIENT_FAILURE	{"system": "grpc", "grpc_log": true}
1.6366781816365309e+09	info	zap/grpclogger.go:92	[core]pickfirstBalancer: UpdateSubConnState: 0xc0000207b0, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial tcp [::1]:6754: connect: connection refused"}	{"system": "grpc", "grpc_log": true}
1.636678181636548e+09	info	zap/grpclogger.go:92	[core]Channel Connectivity change to TRANSIENT_FAILURE	{"system": "grpc", "grpc_log": true}
1.636678181636576e+09	warn	zap/options.go:212	finished client unary call	{"system": "grpc", "span.kind": "client", "grpc.service": "protof.Product", "grpc.method": "search", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:6754: connect: connection refused\"", "grpc.code": "Unavailable", "grpc.time_ms": 0.9290000200271606}
2021/11/11 16:49:41 could not greet: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp [::1]:6754: connect: connection refused"
exit status 1
```

and also note that `grpcZap.ReplaceGrpcLoggerV2(logger)` is a global action.  If we want the detail in the above output we need to expose service wide configuration  to override the grpc logger. 

Kindly advise

Thanks, 
Pavan

Reply:
> also note that grpcZap.ReplaceGrpcLoggerV2(logger) is a global action

I think this is the key. I don't see a way out other than treating it as a collector-wide config, like the main logging level.